src/ssl.c

```diff
@@ -47012,9 +47052,30 @@ WOLFSSL_BN_ULONG wolfSSL_BN_mod_word(const WOLFSSL_BIGNUM *bn,
         return (WOLFSSL_BN_ULONG)WOLFSSL_FATAL_ERROR;
     }
 
-    if (mp_mod_d((mp_int*)bn->internal, w, &ret) != MP_OKAY) {
-        WOLFSSL_MSG("mp_add_d error");
-        return (WOLFSSL_BN_ULONG)WOLFSSL_FATAL_ERROR;
+    if (w <= MP_MASK) {
+        WOLFSSL_BN_ULONG bn_ret;
```

Comment:
Why not use `mp_digit` here and fix uses of `WOLFSSL_BN_ULONG` to define `unsigned long`?

Reply:
Indeed.  See above.